.pre-commit-config.yaml

```diff
@@ -140,7 +140,7 @@ repos:
       - id: markdownlint-fix
         args: [--disable, MD013]
   - repo: https://github.com/pdm-project/pdm
-    rev: 2.16.1
+    rev: 2.17.1
```

Comment:
@gaya3-zipstack @hari-kuriakose we should update to the latest PDM version in all our dev envs and Dockerfiles as well. If not we can downgrade this to `2.16.1` itself

Reply:
Yeah asked the same qn to @hari-kuriakose 
His reply:
PDM version needs to be uniform everywhere.
In that case let's skip this PR for now. Not urgent.
Sometimes the pre-commit agent might even auto correct itself.